### PR TITLE
chore: Migrate to jest

### PR DESCRIPTION
**Pull request by jmelberg-okta**
### Description

Removes outdated dependency on `grunt-contrib-jasmine` to `jest`.

This PR is broken into commits that consist of generated-code and manual code changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "lint": "eslint .",
     "lint:report": "eslint -f checkstyle -o build2/reports/lint/eslint-checkstyle-result.xml .",
     "test": "npm run test:unit",
-    "test:unit": "jest test/*.spec.js --silent",
+    "test:unit": "jest test/spec/ --silent",
```
Comment (haishengwu-okta) [inline package.json]: - how about merge `test` and `test:unit`. seems doesn't need both?
- maybe have another one for ci `test:report: npm run test -- --ci ...`

File: scripts/unit.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 source ${OKTA_HOME}/${REPO}/scripts/setup.sh
 
 export TEST_SUITE_TYPE="jsunit"
-export TEST_RESULT_FILE_DIR="${REPO}/build2/reports/jasmine"
+export TEST_RESULT_FILE_DIR="${REPO}/build2/reports/"
```
Comment (haishengwu-okta) [inline scripts/unit.sh]: - looks like you didn't enable jest-junit yet. would be a result file generated?

File: package.json (hunk 1 above)
Comment (jmelberg-okta) [reply package.json]: Good idea. I had a feeling we'd eventually add `e2e` tests within this repo, but for now we should group these together.

File: scripts/unit.sh (hunk 2 above)
Comment (haishengwu-okta) [reply scripts/unit.sh]: looks like it doesn't match jest junit config `./build2/reports/unit/junit-result.xml`.
or maybe simplify the config to be `./build2/reports/jest-junit-result.xml`.

File: scripts/unit.sh [hunk 3]
```diff
@@ -5,7 +5,7 @@ source ${OKTA_HOME}/${REPO}/scripts/setup.sh
 export TEST_SUITE_TYPE="jsunit"
 export TEST_RESULT_FILE_DIR="${REPO}/build2/reports/"
 
-if ! npm test; then
+if ! npm run test:unit; then
```
Comment (haishengwu-okta) [inline scripts/unit.sh]: `test:report`

File: scripts/unit.sh (hunk 2 above)
Comment (jmelberg-okta) [reply scripts/unit.sh]: We're outputting two separate directories within `build2/reports`:

- `lint`: Results from `npm run lint:report`
- `unit`: Results from `npm run test:report`

I'll update this to be clear which directory we're outputting to:

```
export TEST_RESULT_FILE_DIR="${REPO}/build2/reports/unit"
```

**Pull request merged**